### PR TITLE
Fixes issues with feeds manager reconnection

### PR DESCRIPTION
**Pull request by jkongie**
* Fixes an issue where update fails because a db deadline was not provided
* Fixes an issue where an unconnected feeds manager updated with the correct details would not attempt a connection
* Fixes an issue where a connection would register as connected even if wsrpc is blocking waiting to establish the connection

### CODE REVIEW

File: core/services/feeds/service.go [hunk 1]
```diff
@@ -171,7 +174,7 @@ func (s *service) UpdateFeedsManager(ctx context.Context, mgr FeedsManager) erro
 	}
 
 	if err = s.connMgr.Disconnect(mgr.ID); err != nil {
-		return errors.Wrap(err, "could not restart the connection")
+		logger.Info("[Feeds] Feeds Manager not connected, attempting to connect")
```
Comment (jkongie) [inline core/services/feeds/service.go:177]: An error only occurs here if the connection was not active, so we want to continue with the connection attempt

File: core/services/feeds/service.go [hunk 2]
```diff
@@ -158,6 +158,9 @@ func (s *service) SyncNodeInfo(id int64) error {
 // UpdateFeedsManager updates the feed manager details, takes down the
 // connection and reestablishes a new connection with the updated public key.
 func (s *service) UpdateFeedsManager(ctx context.Context, mgr FeedsManager) error {
+	ctx, cancel := context.WithTimeout(ctx, postgres.DefaultQueryTimeout)
```
Comment (jkongie) [inline core/services/feeds/service.go:161]: Need this to pass the Deadline is set check

File: core/services/feeds/connection_manager.go [hunk 3]
```diff
@@ -159,7 +162,7 @@ func (mgr *connectionsManager) Close() {
 // GetClient returns a single client by id
 func (mgr *connectionsManager) GetClient(id int64) (pb.FeedsManagerClient, error) {
 	conn, ok := mgr.connections[id]
-	if !ok {
+	if !ok || !conn.connected {
```
Comment (connorwstein) [inline core/services/feeds/connection_manager.go:165]: Not sure I follow how this helps since it gets set to true the same time we set connections[id] ?

Comment (jkongie) [reply core/services/feeds/connection_manager.go:165]: It's initially set to false on line 83. It only gets updated to true after the the initial Dial succeeds on line 114

**Review by connorwstein — APPROVED** [review]

**Pull request merged**